doc/0003-wizarding-api.md

```diff
@@ -0,0 +1,227 @@
+# Wizarding API
+
+Date: 2023-07
+
+## Status
+
+Accepted
+
+## Context
+
+Often plugins need a way to display, edit and create SCD elements.
+We want to avoid to put too much effort into implementing the same thing over and over again.
+Also we want to speed up plugin development by extracting some common functionalities out of the plugins.
+
+
+## Decision
+
+We will create a new plugin type, wizard, and provide a wizarding API that enables the followings:
+- Editor plugins can call wizards to display, edit or create specific SCD elements
+- We can provide wizard plugins for specific SCD elements
+- In case there are no wizard plugins installed we will have a simple, generic one
+
+
+### API Specification
+
+In this section we define the lifecycle of a wizard plugin and describe each step.
+
+#### Structure
+
+```txt
+┌──────────────────────────────────┐
+│Core                              │
+│   ┌─────────────┐┌─────────────┐ │
+│   │Editor Plugin││Wizard Plugin│ │
+│   └─────────────┘└─────────────┘ │
+└──────────────────────────────────┘
+```
+
+We separate editors and wizards so that wizards can be used by other editors. 
+This, of course, could create a coupling between plugins, so a default or 
+fallback wizard is necessary to handle all elements e.g.: a generic XML editor
+
+#### Lifecycle
+
+```txt
+┌─────────────────┐ ┌───────────┐┌─────────────────┐     ┃
+│  Editor Plugin  │ │   Core    ││  Wizard Plugin  │     ┃  LEGEND
+└─────────────────┘ └───────────┘└─────────────────┘     ┃  ┌───────────┐
```

Comment:
Does "Editor Plugin" in this case include a menu plugin?

Reply:
no, I was just focusing on the the "editor vs wizard" aspect :)
I could include it if leaving it out only creates confusion